services/blockstorage/service.go

```diff
@@ -38,16 +43,36 @@ type service struct {
 
 	lastCommittedBlock *protocol.BlockPairContainer
 	lastBlockLock      *sync.Mutex
+
+	blockSync *BlockSync
 }
 
-func NewBlockStorage(config Config, persistence adapter.BlockPersistence, stateStorage services.StateStorage, reporting log.BasicLogger) services.BlockStorage {
-	return &service{
+func NewBlockStorage(ctx context.Context, config Config, persistence adapter.BlockPersistence, stateStorage services.StateStorage, gossip gossiptopics.BlockSync, reporting log.BasicLogger) services.BlockStorage {
+	logger := reporting.For(log.Service("block-storage"))
+
+	storage := &service{
 		persistence:   persistence,
 		stateStorage:  stateStorage,
-		reporting:     reporting.For(log.Service("block-storage")),
+		gossip:        gossip,
+		reporting:     logger,
 		config:        config,
 		lastBlockLock: &sync.Mutex{},
 	}
+
+	lastBlock, err := persistence.GetLastBlock()
+
+	if err != nil {
+		logger.Error("could not update last block from persistence", log.Error(err))
+	}
+
+	if lastBlock != nil {
+		storage.updateLastCommittedBlock(lastBlock)
+	}
+
+	storage.blockSync = NewBlockSync(ctx, config, storage, gossip, reporting)
+	gossip.RegisterBlockSyncHandler(storage)
```

Comment:
Change the order of these 2 actions. We should register with gossip first, because the block sync relies on this registration

Reply:
Fixed.